android/app/src/main/java/org/fossasia/openevent/fragments/FeedFragment.java

```diff
@@ -0,0 +1,166 @@
+package org.fossasia.openevent.fragments;
+
+import android.content.SharedPreferences;
+import android.os.Bundle;
+import android.preference.PreferenceManager;
+import android.support.design.widget.Snackbar;
+import android.support.v4.widget.SwipeRefreshLayout;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import org.fossasia.openevent.OpenEventApp;
+import org.fossasia.openevent.R;
+import org.fossasia.openevent.adapters.FeedAdapter;
+import org.fossasia.openevent.api.APIClient;
+import org.fossasia.openevent.data.facebook.FeedItem;
+import org.fossasia.openevent.utils.ConstantStrings;
+import org.fossasia.openevent.utils.NetworkUtils;
+import org.fossasia.openevent.utils.ShowNotificationSnackBar;
+
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+import java.util.List;
+
+import butterknife.BindView;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.schedulers.Schedulers;
+import timber.log.Timber;
+
+/**
+ * Created by rohanagarwal94 on 10/6/17.
+ */
+public class FeedFragment extends BaseFragment {
+
+    private FeedAdapter feedAdapter;
+    private List<FeedItem> feedItems;
+    private SharedPreferences sharedPreferences;
+
+    @BindView(R.id.feed_swipe_refresh) SwipeRefreshLayout swipeRefreshLayout;
+    @BindView(R.id.feed_recycler_view) RecyclerView feedRecyclerView;
+    @BindView(R.id.txt_no_posts) TextView noFeedView;
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        setHasOptionsMenu(true);
+
+        View view = super.onCreateView(inflater, container, savedInstanceState);
+
+        feedItems = new ArrayList<>();
+        RecyclerView.LayoutManager mLayoutManager = new LinearLayoutManager(getContext());
+        feedRecyclerView.setLayoutManager(mLayoutManager);
+        feedAdapter = new FeedAdapter(getContext(), (FeedAdapter.AdapterCallback)getActivity(), feedItems);
+        feedRecyclerView.setAdapter(feedAdapter);
+        sharedPreferences = PreferenceManager.getDefaultSharedPreferences(getContext());
+
+        downloadFeed();
+
+        swipeRefreshLayout.setOnRefreshListener(this::refresh);
+
+        return view;
+    }
+
+    private void downloadFeed() {
+        APIClient.getFacebookGraphAPI()
+                .getPosts(sharedPreferences.getString(ConstantStrings.FACEBOOK_PAGE_ID, null),
+                        getContext().getResources().getString(R.string.fields),
+                        getContext().getResources().getString(R.string.facebook_access_token))
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .doOnComplete(() -> {
```

Comment:
Don't use side effects to trigger things that are to be done when observable completes or fails. Side effects are to be used when -

- Logging the completion or emission
- You don't have access to the subscribe method, so it is a restraint

Furthermore, the doOnError will be called in case of error, but the app will crash with OnErrorNotImplemented because you need to provide onError callback to subscribe method

Since you are subscribing it here, do it like this

```
.subscribe(feed -> { // onNext 
              }, throwable -> { // onError 
              }, () -> { // onComplete 
              });